sky/sdk/lib/widgets/variable_height_scrollable.dart

```diff
@@ -14,18 +12,28 @@ class VariableHeightScrollable extends Scrollable {
   VariableHeightScrollable({
     String key,
     this.builder,
-    this.token
+    this.token,
+    this.layoutState
   }) : super(key: key);
 
   IndexedBuilder builder;
   Object token;
+  BlockViewportLayoutState layoutState;
   bool _contentsChanged = true;
 
+  void initState() {
+    if (layoutState == null)
+      layoutState = new BlockViewportLayoutState();
+    layoutState.addListener(_handleLayoutChanged);
+    super.initState();
+  }
+
   void syncFields(VariableHeightScrollable source) {
     builder = source.builder;
     if (token != source.token)
       _contentsChanged = true;
     token = source.token;
+    layoutState = source.layoutState;
```

Comment:
I was mismanaging this field. For now I'm going to assume that the same value is provided each time the widget is built.  So syncFields()  does ```assert(layoutState == source.layoutState);```. Similarly initState() asserts that layoutState isn't null.